mantidproject/mantid
# Add CMake option to build only the Framework

This is intended to consolidate which `CMakeLists.txt` is used for configuring mantid.

**To test:**

For all of these steps, none of the gui should appear in the output area.
1. Create a fresh build area
2. `cmake -DENABLE_MANTIDPLOT=False ...` with the normal other arguments set including the location of the upper level `CMakeLists.txt`
3. `cmake --build . -- -j <numthread>`
4. `cmake --build . --target FrameworkTests -- -j <numthread>`
5. `ctest -j <numthread>`

_There is no associated issue._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

The additional keyword is ENABLE_MANTIDPLOT and defaults to True.
